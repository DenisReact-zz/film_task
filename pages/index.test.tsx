import Home from "./index";
import { screen, render, fireEvent } from "@testing-library/react";
import "isomorphic-fetch";

describe("Tests for home page", () => {
  beforeEach(() => {
    render(<Home />);
  });
  it("check layout home page", () => {
    const home = screen.getByTestId("home");
    expect(home).toMatchSnapshot();
  });
  it("check arrows of paginaiton", () => {
    const prevArrow = screen.getByTestId("prevArrow");
    const nextArrow = screen.getByTestId("nextArrow");
    const currentPage = screen.getByTestId("currentPage");
    expect(currentPage.textContent).toEqual("1");
    fireEvent.click(nextArrow);
    expect(currentPage.textContent).toEqual("2");
    fireEvent.click(prevArrow);
    expect(currentPage.textContent).toEqual("1");
  });
});
