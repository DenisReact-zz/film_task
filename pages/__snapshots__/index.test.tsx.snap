// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for home page check layout home page 1`] = `
<div
  class="container"
  data-testid="home"
>
  <a
    class="favourite"
    href="/favourites"
  >
    <img
      alt=""
      src="/images/star.svg"
    />
  </a>
  <h2>
    Top 500 films on TMDB
  </h2>
  <div
    class="filter"
    data-testid="filters"
  >
    <div
      class="blocks"
    >
      <button
        data-testid="buttonOfFilter"
      >
        Descending
      </button>
      <button
        data-testid="buttonOfFilter"
      >
        Ascending
      </button>
      <div
        class="thumb"
      />
    </div>
  </div>
  <div
    class="blocks"
  />
  <div
    class="navigation"
  >
    <button
      class="prev"
      data-testid="prevArrow"
    >
      ∟
    </button>
    <p
      data-testid="currentPage"
    >
      1
    </p>
    <button
      class="next"
      data-testid="nextArrow"
    >
      ∟
    </button>
  </div>
</div>
`;
